lock-in device added and updated log.py for multiple logs

debobiswas01 opened the pull request:
SRS_830 device added
added an instrument schema
added lock-in + keithley experiment for oussama chip updated log.py
made all experiments compatible with updated log.py

debobiswas01 committed:
lock-in device added and updated the log.py for multiple logs

SRS_830 device added
added an instrument schema
added lock-in + keithley experiment for oussama chip
updated log.py
made all experiments compatible with updated log.py

comp-phys-marc commented on experiment_configs/lock_in_keithley_gating_oussama.json:
```diff
@@ -0,0 +1,13 @@
+    "sensitivity": 20, 
+    "time_constant": 0, 
+    "slope": 1,
+    "ch_id_1": "R", 
```
If these are already strings then there is no need for `_SRS_lock_in_data_key` above.

comp-phys-marc commented on experiments/keithley_sweep.py:
```diff
@@ -47,8 +47,7 @@ def keithley_sweep(
         log_file,
-        'NIai',
-        'V',
+        [dict(name='NIai', units='V')],
```
I believe this should work but did we test it to make sure? Let's do this before merging.

comp-phys-marc replied:
It works!

comp-phys-marc commented on devices/SRS_830.py:
```diff
@@ -0,0 +1,56 @@
+        file.close()
+        super().__init__('Stanford Lock-in Amplifier SRS 830', dict(interface='GPIB', address='4'), client, schema)
+
+    def set_output_and_readout(self, voltage, frequency, sensitivity, time_constant, slope, reference = 1):
```
let's just be consistent with whitespace throughout the file: `reference=1` :)

comp-phys-marc commented on logging/log.py:
```diff
@@ -4,9 +4,10 @@
 
 class Log:
 
-    def __init__(self, log_file_path, log_name, log_units, params, increment=True):
```
Can we add a test to the test suite to show that this works as intended? What would it look like?

comp-phys-marc commented on devices/SRS_830.py:
```diff
@@ -0,0 +1,56 @@
+import os
+from pathlib import PurePath
```
This file looks great. Can we add a test to the test suite for it?

comp-phys-marc commented on json_schemas/instrument_schemas/SRS_lock_in.json:
```diff
@@ -0,0 +1,54 @@
+{
+    "$id": "json_schemas/instrument_schemas/SRS_lock_in.json",
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "title" : "sourcemeter values validation",
```
Please add an appropriate title.

comp-phys-marc commented on json_schemas/instrument_schemas/SRS_lock_in.json:
```diff
@@ -0,0 +1,54 @@
+{
+    "$id": "json_schemas/instrument_schemas/SRS_lock_in.json",
```
Please name this `SRS_830` instead of `SRS_lock_in` to be consistent with the file name under `devices/`

comp-phys-marc reviewed (changes requested):
Great work. Overall looks solid. Just a few changes please.

comp-phys-marc committed:
Add logger test and catch error in log.py

debobiswas01 committed:
response to feedback

changed the file name to SRS_830

removed the _SRS_lock_in_data_key definition and changed it to a variable instead

created a test for the SRS_830

comp-phys-marc commented:
In the future please don't include a dot in the branch name as this makes it hard to work with in the terminal

debobiswas01 committed:
updated tests for SRS_830

added more thorough test for edge cases of all the parameters

made the spacings beside equals to signs consistent

comp-phys-marc commented on devices/SRS_830.py:
```diff
@@ -4,6 +4,7 @@
 
 
 class SRS830(BaseDevice):
+    _SRS_lock_in_data_key = "channel ID"
```
Why do we add this?

debobiswas01 replied:
I added this here because something similar was added in the NI_DAQ script, so I was trying to stay consistent. If you think it is unnecessary then I can remove it.
![image](https://github.com/Quantum-Science-and-Technology-Lab/labber-wrapper/assets/161180155/5e25e457-8e41-4212-a656-a8c13720ee01)

comp-phys-marc commented on devices/SRS_830.py:
```diff
@@ -50,7 +46,7 @@ def set_output_and_readout(self, voltage, frequency, sensitivity, time_constant,
     def read(self, ch_id_1, ch_id_2):
-        result_1 = self.instr.getValue(self._SRS_lock_in_data_key(ch_id_1))
-        result_2 = self.instr.getValue(self._SRS_lock_in_data_key(ch_id_2))
+        result_1 = self.instr.getValue(self._SRS_lock_in_data_key, ch_id_1)
```
I think it is one in which case we don't want this `_SRS_lock_in_data_key` right?

debobiswas01 committed:
addressing feedback

Commit:
Bug hunting

	- added experiment schema
	- overrode BaseDevice validation (temp)
	- TODO: flush out instrument schema
	- logging bug
	- added validation in lock_in_keithley_gating_oussama
	- changed experiment to a python file

comp-phys-marc commented:
Let's flush out the instrument schema for the SRS before merging.

comp-phys-marc committed:
Finish TODO and cleanup repo

	- made usage of device and instrument terminology consistent
	- flushed out SRS 830 isntrument config

Commit:
Catch device/instrument terminology mismatch, run tests

comp-phys-marc committed:
Fix treatment of SET and AWG_SET as devices

Commit:
Refactor for terminology consistency, run tests

Commit:
Merge branch 'SRS_830-device-added-&-log.py-updated' of https://github.com/Quantum-Science-and-Technology-Lab/labber-wrapper into SRS_830-device-added-&-log.py-updated

comp-phys-marc reviewed (approved):
Looks good!
